### PR TITLE
Enhancement: allowing zero-length static array

### PR DESCRIPTION
**Pull request by ahangsu**
Fixes https://github.com/algorand/js-algorand-sdk/issues/497

### CODE REVIEW

File: .test-env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="master"
+SDK_TESTING_BRANCH="zero-length"
```
Comment (ahangsu) [inline .test-env:3]: TODO: revert before merge into develop
```suggestion
SDK_TESTING_BRANCH="master"
```

**Review by michaeldiamant — APPROVED** [review]
@ahangsu Looks correct - Thanks for bringing the fix into js-algorand-sdk!

File: tests/10.ABI.ts [hunk 2]
```diff
@@ -125,7 +125,6 @@ describe('ABI type checking', () => {
       '[][][]',
       'stuff[]',
       // static array
-      'ufixed32x10[0]',
```
Comment (algochoi) [inline tests/10.ABI.ts:128]: I'd just confirm that something like `bool[01]` also fails for the regex.

Comment (ahangsu) [reply tests/10.ABI.ts:128]: we only allow `bool[0]` and `bool[1]`, but not `bool[01]`, see comment here: https://github.com/algorand/js-algorand-sdk/issues/497#issuecomment-1013530294

Comment (algochoi) [reply tests/10.ABI.ts:128]: Right, I think `bool[01]` should be added here just to check that it correctly throws an error (since it is not allowed).

Comment (ahangsu) [reply tests/10.ABI.ts:128]: got it got it, that's a good point, will add testcase to exclude this case

Comment (ahangsu) [reply tests/10.ABI.ts:128]: just add new testcase here: [c4d93ec](https://github.com/algorand/js-algorand-sdk/pull/698/commits/c4d93ececf418b579ea9c173054c20e4a5acb2ac)

**Review by algochoi — APPROVED** [review]
LGTM!

**Pull request merged**